ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -298,10 +304,13 @@ private ResolvedConfig(final ConfigDef.Type type, final String key, final Object
       )
   );
 
-  private static Optional<ResolvedConfig> resolveStreamsConfig(final String maybePrefixedKey,
-                                                               final Object value) {
+  private static Optional<ConfigValue> resolveStreamsConfig(final String maybePrefixedKey,
+                                                            final Object value) {
     final String key = maybePrefixedKey.startsWith(KSQL_STREAMS_PREFIX)
         ? maybePrefixedKey.substring(KSQL_STREAMS_PREFIX.length()) : maybePrefixedKey;
+    if (key.startsWith(KSQL_CONFIG_PROPERTY_PREFIX)) {
```

Comment:
Basically this means that everything that is not a ksql config property is considered a streams property. If it is further identified as a streams or admin client property, it will have the appropriate type, else will be considered unresolved, correct?

What purpose does the type serve?

Reply:
We need it to determine if any of the configs are secret and obfuscate them if so from the *WithSecretsObfuscated interfaces.